protons/driver.py

```diff
@@ -1287,30 +1392,48 @@ def __init__(self, system, temperature, pH, prmtop, cpin_filename, integrator, p
         self.temperature = temperature
         kT = kB * temperature  # thermal energy
         self.beta = 1.0 / kT  # inverse temperature
+        self.beta_unitless = strip_in_unit_system(self.beta)   # For more efficient calculation of the work (in multiples of KT) during NCMC
         self.pressure = pressure
         self.pH = pH
         self.cpin_filename = cpin_filename
         self.debug = debug
         self.nsteps_per_trial = ncmc_steps_per_trial
+        self.ncmc_prop_per_step = ncmc_prop_per_step
         if implicit:
             self.solvent = "implicit"
         else:
             self.solvent = "explicit"
-        # Create a Verlet integrator to handle NCMC integration
+        # Create a GHMC integrator to handle NCMC integration
         self.compound_integrator = openmm.CompoundIntegrator()
         self.compound_integrator.addIntegrator(integrator)
-        self.ncmc_propagation_integrator = VelocityVerletIntegrator(ncmc_timestep)
+
+        self.ncmc_propagation_integrator = GHMCIntegrator(temperature, 1/units.picosecond, ncmc_timestep, nsteps=self.ncmc_prop_per_step)
```

Comment:
What integrators will be compatible with this integrator? We should make sure to update the tests, and examples to match with an integrator that is compatible.